lib/internal/url.js

```diff
@@ -1504,50 +1502,39 @@ function fileURLToPath(path) {
 // - CR: The carriage return character is also stripped out by the `pathname`
 //       setter.
 // - TAB: The tab character is also stripped out by the `pathname` setter.
-const percentRegEx = /%/g;
-const backslashRegEx = /\\/g;
-const newlineRegEx = /\n/g;
-const carriageReturnRegEx = /\r/g;
-const tabRegEx = /\t/g;
-
 function encodePathChars(filepath) {
-  if (StringPrototypeIncludes(filepath, '%'))
-    filepath = StringPrototypeReplace(filepath, percentRegEx, '%25');
+  filepath = StringPrototypeReplaceAll(filepath, '%', '%25');
   // In posix, backslash is a valid character in paths:
-  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
-    filepath = StringPrototypeReplace(filepath, backslashRegEx, '%5C');
-  if (StringPrototypeIncludes(filepath, '\n'))
-    filepath = StringPrototypeReplace(filepath, newlineRegEx, '%0A');
-  if (StringPrototypeIncludes(filepath, '\r'))
-    filepath = StringPrototypeReplace(filepath, carriageReturnRegEx, '%0D');
-  if (StringPrototypeIncludes(filepath, '\t'))
-    filepath = StringPrototypeReplace(filepath, tabRegEx, '%09');
+  if (!isWindows) filepath = StringPrototypeReplaceAll(filepath, '\\', '%5C');
+  filepath = StringPrototypeReplaceAll(filepath, '\n', '%0A');
+  filepath = StringPrototypeReplaceAll(filepath, '\r', '%0D');
+  filepath = StringPrototypeReplaceAll(filepath, '\t', '%09');
```

Comment:
According to micro-benchmarks, `ReplaceAll` is slower than `Replace`. I prefer using the existing implementation. Referencing: https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v19.md

Reply:
The existing implementation relies on user-mutable `RegExp.prototype[Symbol.replace]`. According to my own research, `replaceAll` on a string was faster than building a robust regex.